### PR TITLE
Milvus: configurable field names

### LINKED ISSUE #1842: [FEATURE] Milvus: configurable field names
**Is your feature request related to a problem? Please describe.**

https://github.com/langchain4j/langchain4j/blob/11855157dd71eb52d603c981a4e3ac561a336107/langchain4j-milvus/src/main/java/dev/langchain4j/store/embedding/milvus/MilvusEmbeddingStore.java

        //static final String ID_FIELD_NAME = "id";
        //static final String TEXT_FIELD_NAME = "text";
        //static final String METADATA_FIELD_NAME = "metadata";
        //static final String VECTOR_FIELD_NAME = "vector";

These are hardcoded, if I want to reuse an existing collection it can have different names for these.   I may also have multiple field names.

**Describe the solution you'd like**

Make these changeable via set, parameter on create or builder

**Describe alternatives you've considered**

Make these changeable via set, parameter on create or builder

Or at least make them properties in a app resources file

**Additional context**
LangChain supports this in Python.

As a Milvus user I usually have different field names and multiple vectors in my collection

### PR DESCRIPTION
**Pull request by hrhrng**
## Issue
Closes #1842 

## Change
Add `FieldDefinition` to hold customized filed name.
Replace default field name with customized filed name.

## General checklist

- [X] There are no breaking changes
- [X] I have added unit and integration tests for my change
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [X] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for changing existing embedding store integration
- [X] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: langchain4j-milvus/src/main/java/dev/langchain4j/store/embedding/milvus/MilvusMetadataFilterMapper.java [hunk 1]
```diff
@@ -15,80 +15,80 @@
 
 class MilvusMetadataFilterMapper {
 
-    static String map(Filter filter) {
+    static String map(Filter filter, String metadataFiledName) {
```
Comment (langchain4j) [inline langchain4j-milvus/src/main/java/dev/langchain4j/store/embedding/milvus/MilvusMetadataFilterMapper.java]:
```suggestion
    static String map(Filter filter, String metadataFieldName) {
```

File: langchain4j-milvus/src/test/java/dev/langchain4j/store/embedding/milvus/MilvusEmbeddingStoreRemovalIT.java [hunk 2]
```diff
@@ -25,6 +25,10 @@ class MilvusEmbeddingStoreRemovalIT extends EmbeddingStoreWithRemovalIT {
             .password(System.getenv("MILVUS_PASSWORD"))
             .consistencyLevel(STRONG)
             .dimension(384)
+            .idFieldName("id_field")
```
Comment (langchain4j) [inline langchain4j-milvus/src/test/java/dev/langchain4j/store/embedding/milvus/MilvusEmbeddingStoreRemovalIT.java]: Why is this change required? I can understand setting these fields explicitly in one of the tests, but this should not be required for all of them

File: langchain4j-milvus/src/test/java/dev/langchain4j/store/embedding/milvus/MilvusEmbeddingStoreIT.java [hunk 3]
```diff
@@ -65,6 +69,10 @@ void should_not_retrieve_embeddings_when_searching() {
                 .consistencyLevel(STRONG)
                 .dimension(384)
                 .retrieveEmbeddingsOnSearch(false)
+                .idFieldName("id_field")
```
Comment (langchain4j) [inline langchain4j-milvus/src/test/java/dev/langchain4j/store/embedding/milvus/MilvusEmbeddingStoreIT.java:72]: Why is this change required? I can understand setting these fields explicitly in one of the tests, but this should not be required for all of them

**Review by ghost — COMMENTED** [review]
@hrhrng thank you!

File: langchain4j-milvus/src/test/java/dev/langchain4j/store/embedding/milvus/MilvusEmbeddingStoreIT.java (hunk 3 above)
Comment (hrhrng) [reply langchain4j-milvus/src/test/java/dev/langchain4j/store/embedding/milvus/MilvusEmbeddingStoreIT.java:72]: The embeddingStore defined in class level share the some instance with the one defined in method level. So their field name must be same.

**Review by ghost — APPROVED** [review]
@hrhrng thank you!

**Pull request merged**